ts/screens/preferences/CalendarScreen.tsx

```diff
@@ -0,0 +1,224 @@
+  Platform,
+  StyleSheet,
+  TouchableHighlight,
+  TouchableOpacity,
+  View
+} from "react-native";
+
+import { Button, Text } from "native-base";
+
+import * as pot from "italia-ts-commons/lib/pot";
+
+import I18n from "../../i18n";
+
+import { Dispatch, ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+
+import TopScreenComponent from "../../components/screens/TopScreenComponent";
+
+import { preferredCalendarSaveSuccess } from "../../store/actions/persistedPreferences";
+
+import IconFont from "../../../ts/components/ui/IconFont";
+
+import RNCalendarEvents, { Calendar } from "react-native-calendar-events";
+
+import customVariables from "../../theme/variables";
+
+const styles = StyleSheet.create({
+  content: {
+    padding: customVariables.contentPadding,
+    paddingTop: 48
+  },
+  calendarItemWrapper: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+    paddingVertical: 16,
+    borderBottomWidth: 1,
+    borderBottomColor: customVariables.brandLightGray,
+    color: customVariables.contentPrimaryBackground
+  },
+  separator: {
+    paddingTop: 25,
+    width: "100%"
+  }
+});
+
+const TouchableComponent =
+  Platform.OS === "ios" ? TouchableHighlight : TouchableOpacity;
+
+type CalendarItemProps = {
+  calendar: Calendar;
+  onPress: () => void;
+};
+
+/**
+ * Renders a Calendar as FlatList item
+ */
+const CalendarItem: React.SFC<CalendarItemProps> = props => (
+  <TouchableComponent onPress={props.onPress}>
+    <View style={styles.calendarItemWrapper}>
+      <Text link={true}>{props.calendar.title}</Text>
+      <IconFont
+        name="io-right"
+        color={customVariables.contentPrimaryBackground}
+      />
+    </View>
+  </TouchableComponent>
+);
+
+type OwnProps = NavigationInjectedProps;
+
+type Props = ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps> &
+  ReduxProps &
+  OwnProps;
+
+type State = {
+  calendars: pot.Pot<ReadonlyArray<Calendar>, ResourceError>;
+};
+
+const INITIAL_STATE: State = {
+  calendars: pot.none
+};
+
+type FetchError = {
+  kind: "FETCH_ERROR";
+};
+
+type ResourceError = FetchError;
+
+const mapResourceErrorToMessage = (resourceError: ResourceError): string => {
+  switch (resourceError.kind) {
+    case "FETCH_ERROR":
+      return I18n.t("messages.cta.errors.fetchCalendars");
+
+    default: {
+      // Exhaustive check
+      return ((): never => resourceError.kind)();
+    }
+  }
+};
+
+/**
+ * Implements the biometric recognition preference screen where the user can
+ * opt for explicitly not using fingerprint. This class shows up two possible
+ * scenarios:
+ * * a switch to enable/disable biometric recognition. Enabled if the
+ *   fingerprint is enrolled.
+ * * the above switch (disabled) with a button below that redirects the user to
+ *   device settings otherwise.
+ *
+ * Please note that this screen must not be displayed if biometric recognition
+ * is unavailable.
+ */
+/**
```

Comment:
is this comment about this class? (then it should be merged with the comment above) or is it misplaced?